tests/unit/modules/test_x509.py

```diff
@@ -186,6 +187,244 @@ def test_create_certificate(self):
+        ca_cert = x509.create_certificate(text=True,
+                                          signing_private_key=ca_key,
+                                          CN='Redacted Root CA',
+                                          O='Redacted',
+                                          C='BE',
+                                          ST='Antwerp',
+                                          L='Local Town',
+                                          Email='certadm@example.org',
+                                          basicConstraints="critical CA:true",
+                                          keyUsage="critical cRLSign, keyCertSign",
+                                          subjectKeyIdentifier='hash',
+                                          authorityKeyIdentifier='keyid,issuer:always',
+                                          days_valid=3650,
+                                          days_remaining=0)
+
+        fmt = '%Y-%m-%d %H:%M:%S'
+        # We also gonna use the current date in UTC format for verification
+        not_before = datetime.datetime.utcnow()
+        # And set the UTC timezone to the naive datetime resulting from parsing
+        not_before = not_before.replace(tzinfo=M2Crypto.ASN1.UTC)
+        not_before_str = datetime.datetime.strftime(not_before, fmt)
+
+        # Sign a new server certificate with the CA
+        server_cert = x509.create_certificate(text=True,
+                                              signing_private_key=ca_key,
+                                              signing_cert=ca_cert,
+                                              CN='Redacted Normal Certificate',
+                                              O='Redacted',
+                                              C='BE',
+                                              ST='Antwerp',
+                                              L='Local Town',
+                                              Email='certadm@example.org',
+                                              basicConstraints="critical CA:false",
+                                              keyUsage="critical keyEncipherment",
+                                              subjectKeyIdentifier='hash',
+                                              authorityKeyIdentifier='keyid,issuer:always',
+                                              not_before=not_before_str,
+                                              days_remaining=0)
+
+        # Save server certificate to disk so we can check its properties
+        with tempfile.NamedTemporaryFile('w+', delete=False) as server_cert_file:
+            server_cert_file.write(salt.utils.stringutils.to_str(server_cert))
+            server_cert_file.flush()
+
+        # Retrieve not_after property from server certificate
+        server_cert_details = x509.read_certificate(server_cert_file.name)
+        not_before_from_cert = server_cert_details['Not Before']
+
+        # Cleanup
+        os.remove(server_cert_file.name)
```

Comment:
can we add a try/finally here and on the other tests to ensure the file gets cleaned up even if anything fails above?

Reply:
That makes sense, I will add it there. Thanks!

Reply:
Ok, so instead of adding a try/finally I decided to rely on with/NamedTemporaryFile and define the comparison variables ahead of time to prevent a "name not defined" error in case there is a problem with the tests. Good/not good?

Reply:
thanks that should work too :)